x/oracle/types/hash.go

```diff
@@ -0,0 +1,124 @@
+package types
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	"gopkg.in/yaml.v3"
+)
+
+var _ yaml.Marshaler = AggregateVoteHash{}
+
+// AggregateVoteHash is hash value to hide vote exchange rates which is
+// formatted as a HEX string:
+// SHA256("{salt}:{symbol}:{exchangeRate},...,{symbol}:{exchangeRate}:{voter}")
+type AggregateVoteHash []byte
+
+// GetAggregateVoteHash computes hash value of ExchangeRateVote to avoid
+// redundant DecCoins stringify operation.
+func GetAggregateVoteHash(salt string, exchangeRatesStr string, voter fmt.Stringer) AggregateVoteHash {
+	hash := tmhash.NewTruncated()
+	sourceStr := fmt.Sprintf("%s:%s:%s", salt, exchangeRatesStr, voter.String())
```

Comment:
If we call `Write()` for every part then we have to call write 4 times. Doesn't feel right. 
Any thoughts?

Reply:
@Tikaryan it will be more performant than casting the bytes to string and then concatenate them. Ugly though, but we write it once :)

I doesn't feel nice, but Go has such interface for any Hasher.